File: refactor_item_ui.js
const fs = require('fs');
const file = 'c:/Users/lenovo/Desktop/Shaj ERP/lib/features/items/presentation/screens/add_edit_item_screen.dart';

// Read file as utf-16le because we know it's UTF-16 LE without BOM
let buffer = fs.readFileSync(file);
let content = buffer.toString('utf16le');

// Check if it's already UTF-8 by looking for null bytes
if (!content.includes('import')) {
  // It was probably utf8 already
  content = buffer.toString('utf8');
}

// 1. Add _buildResponsiveRow helper method if not exists
if (!content.includes('Widget _buildResponsiveRow(List<Widget> children)')) {
  const helperCode = 
  Widget _buildResponsiveRow(List<Widget> children) {
    return LayoutBuilder(
      builder: (context, constraints) {
        if (constraints.maxWidth < 600) {
          // Mobile: Stack vertically
          return Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: children.map((child) => Padding(
              padding: const EdgeInsets.only(bottom: 16.0),
              child: child,
            )).toList(),
          );
        } else {
          // Tablet/Desktop: Row
          List<Widget> rowChildren = [];
          for (int i = 0; i < children.length; i++) {
            rowChildren.add(Expanded(child: children[i]));
            if (i < children.length - 1) {
              rowChildren.add(const SizedBox(width: 16));
            }
          }
          return Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: rowChildren,
          );
        }
      },
    );
  }

  Widget _buildSectionCard({;
  content = content.replace('  Widget _buildSectionCard({', helperCode);
}

// 2. Replace hardcoded Rows with _buildResponsiveRow
const regex2 = /Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*\]\s*,\s*\)/g;
content = content.replace(regex2, (match, p1, p2) => {
  return '_buildResponsiveRow([\n            ' + p1.trim() + ',\n            ' + p2.trim() + ',\n          ])';
});

const regex3 = /Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*\]\s*,\s*\)/g;
content = content.replace(regex3, (match, p1, p2, p3) => {
  return '_buildResponsiveRow([\n            ' + p1.trim() + ',\n            ' + p2.trim() + ',\n            ' + p3.trim() + ',\n          ])';
});

const regex4 = /Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*([\s\S]*?),\s*\),\s*\]\s*,\s*\)/g;
content = content.replace(regex4, (match, p1, p2, p3, p4) => {
  return '_buildResponsiveRow([\n            ' + p1.trim() + ',\n            ' + p2.trim() + ',\n            ' + p3.trim() + ',\n            ' + p4.trim() + ',\n          ])';
});

// We must write it back as utf16le because dart format and the user's IDE expect it?
// Actually, flutter supports UTF-8, but let's just write UTF-8 to fix it forever.
fs.writeFileSync(file, content, 'utf8');
console.log('Done refactoring UI');

File: refactor_item_ui.dart
import 'dart:io';

void main() {
  final file = File('lib/features/items/presentation/screens/add_edit_item_screen.dart');
  var content = file.readAsStringSync();

  // 1. Add _buildResponsiveRow helper method if not exists
  if (!content.contains('Widget _buildResponsiveRow(List<Widget> children)')) {
    final helperCode = '''
  Widget _buildResponsiveRow(List<Widget> children) {
    return LayoutBuilder(
      builder: (context, constraints) {
        if (constraints.maxWidth < 600) {
          // Mobile: Stack vertically
          return Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: children.map((child) => Padding(
              padding: const EdgeInsets.only(bottom: 16.0),
              child: child,
            )).toList(),
          );
        } else {
          // Tablet/Desktop: Row
          List<Widget> rowChildren = [];
          for (int i = 0; i < children.length; i++) {
            rowChildren.add(Expanded(child: children[i]));
            if (i < children.length - 1) {
              rowChildren.add(const SizedBox(width: 16));
            }
          }
          return Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: rowChildren,
          );
        }
      },
    );
  }

  Widget _buildSectionCard({''';
    content = content.replaceAll('  Widget _buildSectionCard({', helperCode);
  }

  // 2. Replace hardcoded Rows with _buildResponsiveRow
  // This is tricky with regex, so we'll do some targeted replacements for known sections.

  // Helper function to replace a specific row block with _buildResponsiveRow
  String replaceRowBlock(String source, String searchRow, String replacement) {
    if (source.contains(searchRow)) {
      return source.replaceAll(searchRow, replacement);
    }
    return source;
  }

  // We will do a generic replacement for common patterns.
  // Pattern: Row( children: [ Expanded( child: Widget1 ), SizedBox(width: 16), Expanded( child: Widget2 ) ] )
  final regex = RegExp(r'Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*\]\s*,\s*\)', dotAll: true);
  
  content = content.replaceAllMapped(regex, (match) {
    final widget1 = match.group(1);
    final widget2 = match.group(2);
    return '_buildResponsiveRow([\n            \,\n            \,\n          ])';
  });

  // Pattern for 3 items: Row( children: [ Expanded( child: Widget1 ), SizedBox(width: 16), Expanded( child: Widget2 ), SizedBox(width: 16), Expanded( child: Widget3 ) ] )
  final regex3 = RegExp(r'Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*\]\s*,\s*\)', dotAll: true);
  
  content = content.replaceAllMapped(regex3, (match) {
    final widget1 = match.group(1);
    final widget2 = match.group(2);
    final widget3 = match.group(3);
    return '_buildResponsiveRow([\n            \,\n            \,\n            \,\n          ])';
  });

  // Pattern for 4 items
  final regex4 = RegExp(r'Row\(\s*children:\s*\[\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*const SizedBox\(width:\s*\d+\),\s*Expanded\(\s*child:\s*(.*?),\s*\),\s*\]\s*,\s*\)', dotAll: true);
  
  content = content.replaceAllMapped(regex4, (match) {
    final widget1 = match.group(1);
    final widget2 = match.group(2);
    final widget3 = match.group(3);
    final widget4 = match.group(4);
    return '_buildResponsiveRow([\n            \,\n            \,\n            \,\n            \,\n          ])';
  });

  file.writeAsStringSync(content);
  print('Refactored AddEditItemScreen to be responsive.');
}
